File: hashtable/1_TwoSum/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> myMap;
        for(int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if ( myMap.find(complement) != myMap.end()) {
                return {myMap[complement], i};
            }
            myMap[nums[i]] = i;
        }
        throw std::runtime_error("There are'nt target in the nums");
    }
};

File: hashtable/1_TwoSum/Solution.go
func twoSum(nums []int, target int) []int {

    buffer := map[int]int{}

    for i := 0; i < len(nums); i++ {
        complement := target - nums[i]
        if val, ok := buffer[complement]; ok {
            return []int{val,i}
        }
        buffer[nums[i]] = i
    }
    return nil
}